### PR TITLE
Implement generic copy color to RDRAM for GLES2 devices

### PR DESCRIPTION
**Pull request by fzurita**
Async copy is not supported for GLES2 devices unless using EGL Image extension which has platform independent implementations.

I will add the EGL Image extension for Android next.

### CODE REVIEW

File: src/Graphics/OpenGLContext/opengl_ColorBufferReaderWithReadPixels.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	for (unsigned int lnIndex = 0; lnIndex < _height; ++lnIndex) {
+		std::copy_n(pixelData.get() + (lnIndex*strideBytes), widthBytes, m_pixelData.data() + lnIndex*widthBytes);
+	}
+	return m_pixelData.data();
+}
+
+void ColorBufferReaderWithReadPixels::cleanUp()
+{
+	glUnmapBuffer(GL_PIXEL_PACK_BUFFER);
```
Comment (loganmc10) [inline src/Graphics/OpenGLContext/opengl_ColorBufferReaderWithReadPixels.cpp]: I believe this should be removed

**Comment by loganmc10** [comment]
Where does it check for _sync? Won't this run even is the user has Asynx copies selected? It should probably only run if Sync is selected, and be disabled otherwise, or there will be performance impacts

**Comment by fzurita** [comment]
It doesn't check for it. My reasoning was that if a user enables copy color buffer to RDRAM, even in sync mode, that they would rather have the option do something rather than nothing at all.

File: src/Graphics/OpenGLContext/opengl_ColorBufferReaderWithReadPixels.cpp (hunk 1 above)
Comment (fzurita) [inline src/Graphics/OpenGLContext/opengl_ColorBufferReaderWithReadPixels.cpp]: Whoops, missed that.

**Comment by loganmc10** [comment]
Ya but Async mode is the default, so by default this is going to cause poor performance, and the user won't know why.

**Comment by fzurita** [comment]
Ok, we could make the default be "off" for GLES 2 devices.

**Comment by gonetz** [comment]
Merged.

**Pull request closed without merging**